### PR TITLE
Chore/use ember namespaces

### PR DESCRIPTION
**Pull request by DominikGuzei**
Improves the namespace function to return the last part of the namespace so the classes etc. can be directly assigned to the return value.

It looks kind of frightening because all files / classes in the project have been changed to use the new namespacing technique.

the upside is approx. 3 lines less per file and less repeating yourself! :-)

### CODE REVIEW

**Comment by tombu** [comment]
Are these 'ember namespaces' ? Because I think if we change the type of namespace now it should be the first an last change. :wink:

File: app/assets/javascripts/helpers/namespace.js.coffee [hunk 1]
```diff
@@ -3,6 +3,8 @@
   namespaces = path.split '.'
   target = window
   
-  for namespace in namespaces
-    target[namespace] or= {} 
-    target = target[namespace]
\ No newline at end of file
+  for namespace, index in namespaces
```
Comment (tombu) [inline app/assets/javascripts/helpers/namespace.js.coffee:6]: We don´t use `index`, do we?

Comment (DominikGuzei) [inline app/assets/javascripts/helpers/namespace.js.coffee:6]: right, this is a leftover from an experiment :-D will delete it

**Comment by DominikGuzei** [comment]
these are not ember namespaces, but as stated in the corresponding issue #71 ember really sucks with namespaces, so we can't use them like we do. It'll probably be the last namespace improvement ;-)

**Comment by stravid** [comment]
Shouldn't it be `class (namespace 'slotcars').Builder` instead of `class namespace('slotcars').Builder`?

I know I'm a bitch :/

**Comment by DominikGuzei** [comment]
I originally wrote it that way but had the feeling it is not as clear as the well known "method call"

**Comment by stravid** [comment]
Well our CoffeeScript style guide says we should change it. And I would say the same, @tombu @bmedicke @MathiasPaumgarten what do you guys say?

**Comment by DominikGuzei** [comment]
ok will change it ;-)

**Comment by roboshoes** [comment]
Well, now you already changed it - which is totally fine - allthough I would have agreed with you. 

Anyways - nice job.

**Pull request merged**